graph/src/data/store/scalar/bigint.rs

```diff
@@ -0,0 +1,391 @@
+use num_bigint;
+use serde::{self, Deserialize, Serialize};
+use stable_hash::utils::AsInt;
+use stable_hash::StableHash;
+use thiserror::Error;
+use web3::types::*;
+
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+use std::ops::{Add, BitAnd, BitOr, Div, Mul, Rem, Shl, Shr, Sub};
+use std::str::FromStr;
+
+pub use num_bigint::Sign as BigIntSign;
+
+use crate::runtime::gas::{Gas, GasSizeOf, SaturatingInto};
+
+// Use a private module to ensure a constructor is used.
+pub use big_int::BigInt;
+mod big_int {
+    use std::{
+        f32::consts::LOG2_10,
+        fmt::{self, Display, Formatter},
+    };
+
+    #[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+    pub struct BigInt(num_bigint::BigInt);
+
+    impl Display for BigInt {
+        fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
+            self.0.fmt(f)
+        }
+    }
+
+    impl BigInt {
+        // Postgres `numeric` has a limit documented here [https://www.postgresql.org/docs/current/datatype-numeric.htm]:
+        // "Up to 131072 digits before the decimal point; up to 16383 digits after the decimal point"
+        // So based on this we adopt a limit of 131072 decimal digits for big int, converted here to bits.
```

Comment:
Maybe 16383 is more suitable limit here, not that any of those is likely to be reached anyway.

Reply:
I'll leave that as a fix for another time ...